### PR TITLE
Add JSON-LD for EIPS

### PR DESCRIPTION
**Pull request by fulldecent**
Extracted from PR at https://github.com/ethereum/EIPs/pull/2738

### CODE REVIEW

File: _layouts/eip.html [hunk 1]
```diff
@@ -42,3 +42,22 @@ <h1 class="page-heading">
...
+{% comment %}
+Article schema specification:
+https://schema.org/Article
+{% endcomment %}
+<script type="application/ld+json">
+  {
+    "@context": "http://schema.org",
+    "@type": "Article",
+    "author": "{% include authorlist.html authors=page.author %}",
+    "name": "EIP-{{ page.eip | xml_escape }}: {{ page.title | xml_escape }}{% if page.status == "Draft" or page.status = "Last Call" %} [DRAFT]{% endif %}",
```
Comment (MicahZoltu) [inline _layouts/eip.html:54]: From CI:
```
Liquid Warning: Liquid syntax error (line 50): Unexpected character = in "page.status == "Draft" or page.status = "Last Call"" in /_layouts/eip.html
```

**Comment by MicahZoltu** [comment]
Same comment as in #2738.  You should get this added to the next EIPIP meeting agenda since it is a meta change.

File: _layouts/eip.html (hunk 1 above)
Comment (fulldecent) [reply _layouts/eip.html]: Corrected at b929e3d

**Comment by fulldecent** [comment]
EIPIP is not authoritative and does not have decision makers. Therefore it is ineffectual and that is why I left.

This current matter affects nobody, it is merely a HTML coding update, I believe it can be accepted with input only by a single person with commit access rather than requiring votes, socializing or other consensus measures.

**Comment by MicahZoltu** [comment]
Took another gander at this one and think it is fine.

**Comment by fulldecent** [comment]
Cool, thank you

**Comment by fulldecent** [comment]
Related: https://github.com/ethereum/EIPs/pull/2811

File: _layouts/eip.html [hunk 2]
```diff
@@ -42,3 +42,22 @@ <h1 class="page-heading">
   {{ content }}
 
 </div>
+{% comment %}
+Article schema specification:
+https://schema.org/Article
```
Comment (axic) [inline _layouts/eip.html:47]: I wonder why was my comment about the schema ignored in the old PR: https://github.com/ethereum/EIPs/pull/2738#discussion_r456624968

I suggested `TechArticle` with more fields.

See `schema.org` for the distinction:
> TechArticle
> Thing > CreativeWork > Article > TechArticle
> A technical article - Example: How-to (task) topics, step-by-step, procedural troubleshooting,  specifications, etc.

vs.

> Article
> Thing > CreativeWork > Article
> An article, such as a news article or piece of investigative report. Newspapers and magazines have articles of many different types and this is intended to cover them all.
>
> See also blog post.

I do not think EIPs should be considered as blog posts or news articles.

**Comment by axic** [comment]
Also as mentioned in https://github.com/ethereum/EIPs/pull/2738#discussion_r456624968 this PR added the JSON+LD as a duplicate entry, given it already exists in `_includes/head.html`.

**Comment by axic** [comment]
[Google's testing tool displays errors](https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Feips.ethereum.org%2FEIPS%2Feip-20):

Screenshot 2020-07-21 at 15 56 23

Can we be a bit more prudent about merging PRs?

**Comment by fulldecent** [comment]
@axic The [JSON-LD in the head](https://github.com/ethereum/EIPs/blob/master/_includes/head.html#L36) is NOT a conflict against this new JSON. The former regards the website a whole (`WebSite`) and the latter regards individual EIPs (`Article` or subclass).

**Comment by fulldecent** [comment]
@axic The Google tool shows errors. And that is fine. We do not currently require images for each EIP and I do not expect that to change soon.

Either way, I believe life is better with this PR merged and if images will be added then that can be considered separately.

SEO was an overarching concern for the EIPs project and this PR is aligned with that goal.

**Comment by fulldecent** [comment]
@axic Regarding `TechArticle` as more appropriate than `Article`. This is a good point, thank you. I support that change here: https://github.com/ethereum/EIPs/pull/2814

**Comment by MicahZoltu** [comment]
> Can we be a bit more prudent about merging PRs?

I contend that this PR was an improvement, thus worth merging.  It added JSON-LD content that improved SEO.  I *believe* the errors/warnings are just "you can do better" things, and not all of them are *possible* for us to address (like the `image` tag, and probably `publisher` tag).

That being said, it would be nice to get the rest added like `headline` (EIP title), `datePublished` (same as created), `dateModified` (ideally the commit timestamp of the last commit to touch the EIP, but if not just `datePublished` perhaps), `mainEntityOfPage` (I'm not sure what this is, but presumably we have one!).

**Comment by fulldecent** [comment]
Other items in this PR -> https://github.com/ethereum/EIPs/pull/2816

Probably Google is flagging as "error" when warning is appropriate. But still, it's Google, and I used to work there, so --\_0_/-- https://github.com/ethereum/EIPs/pull/2816

**Pull request merged**